src/python/grpcio/grpc/_server.py

```diff
@@ -719,6 +708,26 @@ def cancel_all_calls_after_grace():
   return shutdown_event
 
 
+def _join_server_thread(timeout, state):
+  if not timeout:
+    _stop(state, 2.0).wait()
```

Comment:
Magic number. To best emulate "wait as long as it takes", shouldn't the value passed here be astronomically high (and, sadly, an equally magic number)?